File: 3463-AlternatingGroupsI/3463-AlternatingGroupsI.py
# Last updated: 7/31/2025, 5:11:29 PM
class Solution:
    def numberOfAlternatingGroups(self, colors: List[int]) -> int:
        n=len(colors)
        count=0
        for i in range(n):
            a=colors[i]
            b=colors[(i+1)%n]
            c=colors[(i+2)%n]
            if a!=b and b!=c:
                count+=1
        return count
        

File: 3463-AlternatingGroupsI/3463-AlternatingGroupsI.java
// Last updated: 7/31/2025, 5:11:01 PM
class Solution {
    public int numberOfAlternatingGroups(int[] colors) {
        int n=colors.length;
        int count=0;
        for (int i=0;i<n;i++){
            int a=colors[i];
            int b=colors[(i+1)%n];
            int c=colors[(i+2)%n];
            if (a!=b && b!=c){
                count++;
            }
        }  
        return count;
        
    }
}